File: GitToolBox/src/main/java/zielu/gittoolbox/ui/blame/LineInfo.java
package zielu.gittoolbox.ui.blame;

import com.intellij.openapi.editor.Document;
import com.intellij.openapi.editor.Editor;
import com.intellij.openapi.vfs.VirtualFile;
import org.jetbrains.annotations.NotNull;

class LineInfo {
  final VirtualFile file;
  final Editor editor;
  final Document document;
  final int lineIndex;
  final boolean lineModified;
  final int generation;

  LineInfo(@NotNull VirtualFile file, @NotNull Editor editor, @NotNull Document document,
           int lineIndex, int generation) {
    this.file = file;
    this.editor = editor;
    this.document = document;
    this.lineIndex = lineIndex;
    this.generation = generation;
    lineModified = document.isLineModified(lineIndex);
  }
}

File: src/main/kotlin/zielu/gittoolbox/ui/blame/LineInfo.kt
package zielu.gittoolbox.ui.blame

import com.intellij.openapi.editor.Document
import com.intellij.openapi.editor.Editor
import com.intellij.openapi.vfs.VirtualFile

internal data class LineInfo(
  val file: VirtualFile,
  val editor: Editor,
  val document: Document,
  val index: Int,
  val generation: Int,
  val modified: Boolean
) {
  constructor(
    file: VirtualFile,
    editor: Editor,
    document: Document,
    index: Int,
    generation: Int
  ) : this(file, editor, document, index, generation, document.isLineModified(index))
}
